### PR TITLE
Adds public methods that allowed me to display an eml file in a brower

### PR DESCRIPTION
**Pull request by acolchagoff**

### CODE REVIEW

**Comment by acolchagoff** [comment]
hey @barthez what do you think?

**Comment by barthez** [comment]
Hey @denodster, thanks for pinging me, I didn't get the notification about the PR, sorry 😞. Let me look at it over the weekend.

**Comment by acolchagoff** [comment]
okay cool, i'm glad I got your attention. To give you some context, we're using your library to display emails in a browser. The reason I added the attachment methods is because I needed a way to make the page display attached images as well as hotlink to attached files. Therefore I have a rails endpoint that gets attachments by filename. The bug fix is something I ran into in production as we are using my version on our web server and one of our clients uploaded a bad eml file that was missing it's attached images, thus I created a fallback to set src to "" when the image does not appear to be present.

File: lib/eml2html/converter.rb [hunk 1]
```diff
@@ -94,7 +122,13 @@ def read_attachments
     def replace_images_src(html)
       html.gsub(/(?<=src=['"])cid:[^'"]+(?=['"])/) do |match|
         cid = match.sub(/^cid:/, '')
-        @attachments.find{|a| a.cid == cid}.name
+        @cids_in_body.push(cid)
+        attachment = @attachments.find{|a| a.cid == cid }
```
Comment (barthez) [inline lib/eml2html/converter.rb]: If you are using new Ruby you can use save navigator operator:
@attachments.find{|a| a.cid == cid }&.name || ''

File: lib/eml2html/converter.rb [hunk 2]
```diff
@@ -55,6 +56,33 @@ def save_zip(options = {})
       end
     end
 
+    def get_attachments(include_in_doc=false)
+      attachments = []
```
Comment (barthez) [inline lib/eml2html/converter.rb]: Please don't call methods like `get_`. Please call it `attachments`

```ruby
return @attachments if include_in_doc

@attachments.reject do |attachment|
  @cids_in_body.include?(a.cid)
end
```

File: lib/eml2html/converter.rb [hunk 3]
```diff
@@ -55,6 +56,33 @@ def save_zip(options = {})
...
+        end
+      end
+      attachments
+    end
+
+    def get_html
+      html_body
+    end
+
+    def get_attachment(file_name)
```
Comment (barthez) [inline lib/eml2html/converter.rb]: Same as above, don't call it `get_attachment`. `attachment_by_name` would be more appropriate.
```ruby
@attachments.find{|a| a.name == file_name }&.content
```

File: lib/eml2html/converter.rb [hunk 4]
```diff
@@ -55,6 +56,33 @@ def save_zip(options = {})
...
+        @attachments.each do |a|
+          unless @cids_in_body.includes(a.cid)
+            attachments.push(a)
+          end
+        end
+      end
+      attachments
+    end
+
+    def get_html
```
Comment (barthez) [inline lib/eml2html/converter.rb]: I don't see any point of having this.

**Review by barthez — COMMENTED** [review]
Could you please add a test case for new methods you added? Add a fixture file with 2 attachments, one referenced in body, one not. Add some simple test for your new methods, please.

At the end squash your commits into one, with nice commit message, please.

I can release it to RubyGems when you address all the issues.

File: lib/eml2html/converter.rb (hunk 2 above)
Comment (acolchagoff) [reply lib/eml2html/converter.rb]: done

File: lib/eml2html/converter.rb (hunk 3 above)
Comment (acolchagoff) [reply lib/eml2html/converter.rb]: done

File: lib/eml2html/converter.rb (hunk 4 above)
Comment (acolchagoff) [reply lib/eml2html/converter.rb]: I'll just make html_body public, and then we wont need this.

File: lib/eml2html/converter.rb (hunk 1 above)
Comment (acolchagoff) [reply lib/eml2html/converter.rb]: that's pretty cool, and yes, we're using a new enough ruby to take advantage of it, so I'll make the change.

Comment (acolchagoff) [reply lib/eml2html/converter.rb]: are you sure returning nil inside the gsub closure won't cause any bugs?

**Comment by acolchagoff** [comment]
@barthez made some changes, let me know what you think.

**Comment by acolchagoff** [comment]
Oh, I just saw your other comment, I'll do that stuff too.

File: lib/eml2html/converter.rb (hunk 1 above)
Comment (barthez) [reply lib/eml2html/converter.rb]: Yes, `nil` is fine, just tested it:
```
> "asd".gsub(/./) { nil }
 => ""
```

**Comment by barthez** [comment]
@denodster I forgot I asked you for tests too 😬 

Add a separate PR please if you have a will and time.

Also, I thought I will use GH functionality to `squash and merge` and I messed up 🤦‍♂ 

Nevertheless, eml2html 0.0.4 is released.

**Pull request merged**